cachito/errors.py

```diff
@@ -33,6 +33,10 @@ class UnknownHashAlgorithm(CachitoError):
     """The hash algorithm is unknown by Cachito."""
 
 
+class GitError(CachitoError):
+    """An error was encountered during manipulation with a Git repository."""
```

Comment:
Would any of the error types used in [scm.py](https://github.com/containerbuildsystem/cachito/blob/master/cachito/workers/scm.py) be applicable to the checkout error? That's the module where most git operations happen, it might be slightly confusing that GitError is not used there. But if none of the existing error types would be appropriate adding a new one is OK

Reply:
Hmm, errors in scm.py are quite specific so my thinking was to not confuse anyone by misleading them from the root cause. Overall, I think keeping `GitError` would be the best option, but bellow is what I considered so you can make your own judgment.

Looking into documentation why `Repo` object construction can fail

- `NoSuchPathError` - Thrown if a path could not be access by the system
- `InvalidGitRepositoryError` - Thrown if the given repository appears to have an invalid format.

maybe these two could be raised as `RepositoryAccessError`.

Checkout statement may raise

- `exc.CheckoutError` – If at least one file failed to be checked out. This is a summary, hence it will checkout as many files as it can anyway. If one of files or directories do not exist in the index ( as opposed to the original git command who ignores them ). 
- Raise `GitCommandError` if error lines could not be parsed - this truly is an exceptional state

These two could be raised as  `SubprocessCallError` or create a new one with the different name? Or keeping their names? Though if I'm not mistaken, it's a good practice to wrap exceptions from imported libraries into you're own.

Reply:
Hmm, yeah RepositoryAccessError doesn't seem that accurate

+1 to keep GitError